pkg/objectio/meta.go

```diff
@@ -0,0 +1,92 @@
+// | TableID(8B) | SegmentID(8B) | BlockID(8B)  | ColumnCnt(2B) | Chksum(4B) |  Reserved(33B)    |
+// +-------------+---------------+--------------+---------------+------------+-------------------+
+// |                                         ColumnMeta                                          |
+// +---------------------------------------------------------------------------------------------+
+// |                                         ColumnMeta                                          |
+// +---------------------------------------------------------------------------------------------+
+// |                                         ColumnMeta                                          |
+// +---------------------------------------------------------------------------------------------+
+// |                                         ..........                                          |
+// +---------------------------------------------------------------------------------------------+
+// Header Size = 64B
+// TableID = Table ID for Block
+// SegmentID = Segment ID
+// BlockID = Block ID
+// ColumnCnt = The number of column in the block
+// Chksum = Block metadata checksum
+// Reserved = 41 bytes reserved space
+type BlockMeta struct {
+	header *BlockHeader
+	//columns []*ColumnMeta
+}
+
+type BlockHeader struct {
+	tableId     uint64
+	segmentId   uint64
+	blockId     uint64
+	columnCount uint16
+	checksum    uint32
+}
+
+// +---------------------------------------------------------------------------------------------------------------+
+// |                                                    ColumnMeta                                                 |
+// +--------+-------+----------+--------+---------+--------+--------+------------+---------+-----------+-----------+
+// |Type(1B)|Idx(2B)|Offset(4B)|Size(4B)|oSize(4B)|Min(32B)|Max(32B)|BFoffset(4b)|BFlen(4b)|BFoSize(4B)|Chksum(4B) |
+// +--------+-------+----------+--------+---------+--------+--------+------------+---------+-----------+-----------+
+// |                                                    Reserved(33B)                                              |
+// +---------------------------------------------------------------------------------------------------------------+
+// ColumnMeta Size = 128B
+// Type = Metadata type, always 0, representing column meta, used for extension.
+// Idx = Column index
+// Algo = Type of compression algorithm for column data
+// Offset = Offset of column data
+// Size = Size of column data
+// oSize = Original data size
+// Min = Column min value
+// Max = Column Max value
+// BFoffset = Bloomfilter data offset
+// Bflen = Bloomfilter data size
+// BFoSize = Bloomfilter original data size
+// Chksum = Data checksum
+// Reserved = 33 bytes reserved space
+type ColumnMeta struct {
+	typ         uint8
+	idx         uint16
+	alg         uint8
+	location    Extent
+	zoneMap     *index.ZoneMap
```

Comment:
If ZoneMap size if fixed, why not just put it here without a pointer?

Reply:
Basically, the really easy of Marshal/Unmarshal is no mashalling code at all.    
Just cast the struct to a byte slice and do your I/O.   Can we do that?